broker/src/main/java/org/apache/rocketmq/broker/controller/ReplicasManager.java

```diff
@@ -322,35 +438,150 @@ private boolean registerBrokerToController() {
+            return true;
+        } catch (Exception e) {
+            LOGGER.error("update and persist temp broker metadata file failed", e);
+            this.tempBrokerMetadata.clear();
+            return false;
+        }
+    }
+
+    /**
+     * Send applyBrokerId request to controller
+     * @return whether controller has assigned this brokerId for this broker
+     */
+    private boolean applyBrokerId() {
+        try {
+            ApplyBrokerIdResponseHeader response = this.brokerOuterAPI.applyBrokerId(brokerConfig.getBrokerClusterName(), brokerConfig.getBrokerName(),
+                    tempBrokerMetadata.getBrokerId(), tempBrokerMetadata.getRegisterCheckCode(), this.controllerLeaderAddress);
+            return true;
+
+        } catch (Exception e) {
+            LOGGER.error("fail to apply broker id: {}", e, tempBrokerMetadata.getBrokerId());
+            return false;
+        }
+    }
+
+    /**
+     * Create metadata file and delete temp metadata file
+     * @return whether process success
+     */
+    private boolean createMetadataFileAndDeleteTemp() {
+        // create metadata file and delete temp metadata file
+        try {
+            this.brokerMetadata.updateAndPersist(brokerConfig.getBrokerClusterName(), brokerConfig.getBrokerName(), tempBrokerMetadata.getBrokerId());
+            this.tempBrokerMetadata.clear();
+            this.brokerId = this.brokerMetadata.getBrokerId();
+            return true;
+        } catch (Exception e) {
+            LOGGER.error("fail to create metadata file", e);
+            this.brokerMetadata.clear();
+            return false;
+        }
+    }
+
+    /**
+     * Send registerSuccess request to inform controller that now broker has been registered successfully and controller should update broker ipAddress if changed
+     * @return whether request success
+     */
+    private boolean registerSuccess() {
+        try {
+            RegisterSuccessResponseHeader response = this.brokerOuterAPI.registerSuccess(brokerConfig.getBrokerClusterName(), brokerConfig.getBrokerName(), brokerId, localAddress, controllerLeaderAddress);
+            final Long masterBrokerId = response.getMasterBrokerId();
+            final String masterAddress = response.getMasterAddress();
+            if (masterBrokerId == null) {
+                return true;
+            }
+            if (this.brokerId.equals(masterBrokerId)) {
+                changeToMaster(response.getMasterEpoch(), response.getSyncStateSetEpoch());
+            } else {
+                changeToSlave(masterAddress, response.getMasterEpoch(), masterBrokerId);
+            }
+            brokerController.setIsolated(false);
+            return true;
+        } catch (Exception e) {
+            LOGGER.error("fail to send registerSuccess request to controller", e);
+            return false;
+        }
+    }
```

Comment:
registerSuccess method name、this.brokerOuterAPI.registerSuccess、 and request code  REGISTER_SUCCESS can rename. These names give me the understanding that the registration is successful

Reply:
> 

Yep! I rename it to `registerBrokerToController`, only if this request is successful will this register process is completely finished !